packages/code-du-travail-frontend/src/outils/indemniteLicenciement/index.js

```diff
@@ -1,124 +1,113 @@
 import React from "react";
 import PropTypes from "prop-types";
 import { Alert, Button, Container } from "@cdt/ui";
-
+import dynamic from "next/dynamic";
 import getIndemnite from "./indemnite";
 import { Stepper } from "./Stepper";
-import { Anciennete } from "./Anciennete";
-import { FauteGrave } from "./FauteGrave";
-import { Salaire } from "./Salaire";
-import { Primes } from "./Prime";
-import { DateFinContrat } from "./DateFinContrat";
+import { initialData, steps } from "./cas_general";
 import { ResultDetail } from "./ResultDetail";
 
-export const inputStyle = {
-  padding: 0,
-  fontSize: "1.5rem",
-  textAlign: "center"
-};
-
-const INITIAL_STATE = {
-  fauteGrave: false,
-  anciennete: 0,
-  primes: 0,
-  dateFinContrat: "",
-  salaires: Array.from({ length: 12 }).fill(1498.47) // Prefill with SMIC brut
-};
+const containerRef = React.createRef();
 
 class CalculateurIndemnite extends React.Component {
   static propTypes = {
     q: PropTypes.string
   };
 
-  state = INITIAL_STATE;
+  state = {
+    ...initialData,
+    steps: [...steps]
+  };
+  ResultCC = dynamic(import(`./ccn/Result_0044`));
+
+  componentOnChange = ({ key, value }) => {
+    this.setState({
+      [key]: value
+    });
+    if (key === "convention") {
+      if (value.hasCC) {
+        const { ccId } = value;
+        import(`./ccn/${ccId}`).then(module => {
+          this.setState({
+            steps: steps.concat(module.steps),
+            calculConvention: module.getIndemnite,
+            ...module.initialData
+          });
+        });
+        this.ResultCC = dynamic(import(`./ccn/Result_${ccId}`));
+      } else {
+        this.setState({
+          steps: steps.slice(),
+          calculConvention: null
+        });
+      }
+    }
+  };
+
+  resetState = callback => {
+    const initialkeys = Object.keys(initialData);
+    const newState = Object.keys(this.state).reduce((state, key) => {
+      if (initialkeys.indexOf(key) === -1) {
+        state[key] = undefined;
+      } else {
+        state[key] = initialData[key];
+      }
+      return state;
+    }, {});
+    this.setState(newState, callback);
+  };
 
   render() {
     const indemniteData = getIndemnite(this.state);
+
+    const hasIndemniteCC =
+      indemniteData.calculCC && indemniteData.calculCC.indemnite;
+
+    const showResult = indemniteData.errors.length === 0;
+    let ResultComponent;
```

Comment:
conventionnal nitpick: ResultComponent is an instance, so `resultComponent` ?

or 

`ResultComponent = () => <th...` and `{showResult && <ResultComponent/>}`